x-pack/plugins/alerting/server/task_runner/action_scheduler/get_summarized_alerts.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ALERT_UUID } from '@kbn/rule-data-utils';
+import { createTaskRunError, TaskErrorSource } from '@kbn/task-manager-plugin/server';
+import { GetSummarizedAlertsParams, IAlertsClient } from '../../alerts_client/types';
+import {
+  AlertInstanceContext,
+  AlertInstanceState,
+  CombinedSummarizedAlerts,
+  RuleAlertData,
+} from '../../types';
+
+interface GetSummarizedAlertsOpts<
+  State extends AlertInstanceState,
+  Context extends AlertInstanceContext,
+  ActionGroupIds extends string,
+  RecoveryActionGroupId extends string,
+  AlertData extends RuleAlertData
+> {
+  alertsClient: IAlertsClient<AlertData, State, Context, ActionGroupIds, RecoveryActionGroupId>;
+  queryOptions: GetSummarizedAlertsParams;
+}
+
+export const getSummarizedAlerts = async <
```

Comment:
this was a private function in `ExecutionHandler` class. Moved to a reusable helper function